WarehouseProject/src/warehouse/app/WarehouseDB.java

```diff
@@ -173,12 +130,20 @@ public int getSize() {
 				++count;
 			}
 
-			connect.close();
 			return count;
+			connect.close();
 
 		} catch (SQLException e) {
 			System.err.println(e.getMessage());
-			return count;
 		}
 	}
-}
+
+	private void connect() {
+
+		try {
+			connect = DriverManager.getConnection("jdbc:mysql://localhost:3306/employee_data", "root", "");
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+}
```

Comment:
Do you see the red circle with a line in it? This means the file is missing a blank line at the EOF (end of file). 

Go ahead and add the blank line, and title the commit "SQUASH - add EOF blank line"

Then we are going to squash this commit into the previous one as it's not needed on its own. We will do this by doing an interactive rebase. 

A rebase essentially means we will rewrite git history. Any time you do a successful rebase that rewrites history, you will have to add the `--force` flag to your `git push ...` command. This means you are forcing the rewrite, and this should be done at the end of a PR as rewriting `git` history can mess up PR comments and doing this while in the middle of discussion can be seen as bad manners.

First, we will ensure our branch is up to date with `main`. This is a solo project so I am sure it may already be, but this will be good practice

```
git remote update --prune && git merge origin/main
```
`git remote update --prune` safely cleans outdated branches, and `git merge origin/main` will merge the current `main` branch into yours. You want to do this before the rebase as the rebase will remove the merge commit you would hypothetically make if you branch wasn't already up tp date. If this says "Already up to date", you are good to move on. If a "Merge ...." prompt comes up, merge the commit by typing `:wq` in the terminal to save the commit and them move on.

Then we will start the rebase by running 
```
git rebase -i origin/master
```
The `-i` flag means the rebase is "interactive", meaning that the commits will be listed and allow you to edit, squash, delete them, etc.

On the line with the "SQUASH - ... " commit, change the word `pick` to the letter `s`. This means we are going to squash this commit into the previous one. To continue on, type in `:wq` outside of "insert" mode and hit enter. 

Then a prompt with lines and commit messages should show up. Delete everything related to the "SQUASH - ... " commit by hitting the letter `d` twice for each line while the editor isn't in "insert" mode.

Note: lines starting with a `#` are comments and do not have to be deleted.

Once everything but the commit message of the commit we are keeping is deleted, save the edit by typing `:wq` out of "insert" mode and hit enter. If there are no conflicts (and there may be pending on how your other commits are structured), the rebase should succeed and you will be ready to push.

After rebasing, push your branch but with the `--force` flag `git push origin --force main ... ` and see if that pushes to GitHub successfully.